gremlin_serialize_test.go

```diff
@@ -0,0 +1,333 @@
+package gremlin
+
+import (
+	"fmt"
+	"testing"
+)
+
+func makeDummyVertexProperty(label string, value interface{}) VertexProperty {
+	return VertexProperty{
+		Type: "g:VertexProperty",
+		Value: VertexPropertyValue{
+			ID: GenericValue{
+				Type:  "Type",
+				Value: 1,
+			},
+			Value: value,
+			Label: label,
+		},
+	}
+}
+
+func makeDummyVertex(vertexID, vertexLabel string, params map[string]interface{}) Vertex {
+	properties := make(map[string][]VertexProperty)
+	for label, value := range params {
+		var vp []VertexProperty
+		vSlice, err := value.([]interface{})
+		if err {
+			for _, p := range vSlice {
+				vertexProperty := makeDummyVertexProperty(label, p)
+				vp = append(vp, vertexProperty)
+			}
+		} else {
+			vertexProperty := makeDummyVertexProperty(label, value)
+			vp = append(vp, vertexProperty)
+		}
+		properties[label] = vp
+	}
+	vertexValue := VertexValue{
+		ID:         vertexID,
+		Label:      vertexLabel,
+		Properties: properties,
+	}
+	return Vertex{
+		Type:  "g:Vertex",
+		Value: vertexValue,
+	}
+}
+
+func makeDummyProperty(label string, value interface{}) EdgeProperty {
+	return EdgeProperty{
+		Type: "g:Property",
+		Value: EdgePropertyValue{
+			Value: value,
+			Label: label,
+		},
+	}
+}
+
+func makeDummyEdge(edgeID, edgeLabel, inVLabel, outVLabel, inV, outV string, params map[string]interface{}) Edge {
+	properties := make(map[string]EdgeProperty)
+	for label, value := range params {
+		properties[label] = makeDummyProperty(label, value)
+	}
+	edgeValue := EdgeValue{
+		ID:         edgeID,
+		Label:      edgeLabel,
+		InVLabel:   inVLabel,
+		OutVLabel:  outVLabel,
+		InV:        inV,
+		OutV:       outV,
+		Properties: properties,
+	}
+	return Edge{
+		Type:  "g:Edge",
+		Value: edgeValue,
+	}
+}
+
+func makeDummyGenericValue(gvType string, value interface{}) GenericValue {
+	return GenericValue{
+		Type:  gvType,
+		Value: value,
+	}
+}
+
+func edgesMatch(edge1, edge2 Edge) bool {
```

Comment:
This is really handy.  I think it should be pulled out of the test into a more generic utilities file.  Same with vertexesMatch